RA2-138: Homepages (FE part)

SARodrigues opened the pull request:
Fetch and implement the different sections on the homepage.

**In this PR:** 

- Split the Homepage's sections into separate components 
  _(Intro, Journeys, Section)_
- Added a new `homepage` state module to retrieve the homepage's data from its respective API endpoint  
  _Both the homepage attributes (intro) and the sections' information_  
- Added homepage types in use  
- Generic "Section" component allows for:
  - Sections with centered content, image as cover  
  - Sections with left aligned content, image on the right  
  - Sections with right aligned content, image on the left
- Homepage Journey section slides no longer hardcoded  
  _They are now fetched from the API_

**css**  
  - Reworked CSS for new/refactored components  
  - Minor section improvements for the sections on mobile  
  - Minor journeys slider improvements for the slider on mobile

**e2e testing**  
  - Ordering/positioning of the different sections  
  - Intro
    - Intro Content
    - Background image & respective credits  
  - Journeys section  
    - Correct number of sections (if any)
    - Section content
    - Slider's slides' content  
  - Sections (generic)  
    - Correct number of sections (if any)
    - Sections' content
    - Background color   
    - Background image when used as a background
    - Image positioning when left or right aligned
    - Images' credits  

## Testing instructions

- With the current API response, the homepage should look virtually unchanged  
- Journeys' section data should now be coming from the API, and link to the correct pages  
- Sections should be displayed with the API endpoint data  
- Section ordering respects the positioning coming from the API

## Tracking

https://vizzuality.atlassian.net/browse/RA2-138

SARodrigues commented on frontend/tsconfig.tsbuildinfo:
This file seems to be generated automatically. Is that expected?

Bluesmile82 reviewed (commented):
Great job! But could you check the different sizes on responsive. It seems the screens grow wider than the width and we have to scroll to see it all. This could give problems on mobile and big screens

SARodrigues commented:
@Bluesmile82 

> could you check the different sizes on responsive. It seems the screens grow wider than the width and we have to scroll to see it all. This could give problems on mobile and big screens

It looks like on very small screens, due to the original padding/spacing the content would not fit. When using the `responsive` tab on Chrome (or rotating the screen on a physical device), the width of the page would be larger than expected, causing the issue you've found.

**Steps to mitigate the mobile issue:** 
- Tweaked the spacing on the homepage sections for mobile devices  
- Because the Journeys slider would still not display, I tweaked it a bit so it'd display on mobile devices in a useable way as well  
- Made it so the navigation bar won't overlap with the intro/hero section (replicated what is implemented in other pages. We can change this in the future)

With the added tweaks, the homepage should be useable on devices down to 320px width (iPhone 5). 

**Other changes since the review:**  
- I've merged `develop`, which included the new `JourneySlider` component  
- Also refactored the homepage tests to account for the slider's changes  
- Added `tsconfig.tsbuildinfo` to `.gitignore`

**Other notes:**  
I could not get translations to work (current failing test). Will investigate, but most likely will need help.
